tools/report-converter/codechecker_report_converter/spotbugs/output_parser.py

```diff
@@ -131,6 +136,13 @@ def __parse_bug(self, bug):
 
             if event:
                 events.append(event)
+                # Take the line of the innermost block
+                if event.line > backup_line:
+                    backup_line = event.line
+
+        # If <SourceLine> did not contain 'start' attribute, take the backup.
+        if line is None:
+            line = backup_line
```

Comment:
We can simplify the whole thing in the following way, so there is no need for the `backup_line` variable above:
```suggestion
        # If <SourceLine> did not contain 'start' attribute, take it from
        # the last event which line is non zero.
        if line is None:
            line = next(e.line for e in reversed(events) if e.line > 0)
```